.github/workflows/causal_lm_cpp.yml

```diff
@@ -212,6 +212,41 @@ jobs:
           source ./ov/setupvars.sh
           convert_tokenizer ./notus-7b-v1/pytorch/dldt/FP16/ --output ./notus-7b-v1/pytorch/dldt/FP16/ --with-detokenizer --trust-remote-code
           timeout 50s ./build/beam_search_causal_lm ./notus-7b-v1/pytorch/dldt/FP16/ 69 > ./pred.txt
-
-
-          
+  cpp-speculative_decoding_lm-ubuntu:
+    runs-on: ubuntu-20.04-16-cores
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+      - name: Install OpenVINO
+        run: |
+          mkdir ./ov/
+          curl https://storage.openvinotoolkit.org/repositories/openvino/packages/2023.3/linux/l_openvino_toolkit_ubuntu20_2023.3.0.13775.ceeafaf64f3_x86_64.tgz | tar --directory ./ov/ --strip-components 1 -xz
+          sudo ./ov/install_dependencies/install_openvino_dependencies.sh
+      - name: Download, convert and build
+        run: |
+          source ./ov/setupvars.sh
+          python -m pip install --upgrade-strategy eager "optimum>=1.14" -r ./llm_bench/python/requirements.txt "transformers<4.38" ./thirdparty/openvino_tokenizers/[transformers] --extra-index-url https://download.pytorch.org/whl/cpu
```

Comment:
I believe this test should take a lot of time. We could reduce it by changing models if it is really a bottleneck. @pavel-esir , can you please comment on the execution time?

Reply:
![image](https://github.com/openvinotoolkit/openvino.genai/assets/2566854/ce2bfa1d-314c-4a71-bf3e-60f1549cb89e)

Reply:
i agree that these models are a bit heavy, i tried to replace them with lighter combinations e.g. TinyLlama & Llama-7b but Llama cannot be downloaded without access token so it's unavailable at pre-commit.

I also tried with TinnyLlama and OpenLLaMA but surprisingly they have different tokenizers. So the current combination is the lightest i was able to make work, unfortunately there are no lightest Dollies to use as a draft.

Download & conversion of both models take ~10 m, run ~2 m, it's around twice of the time of usual tests here ~5. But even without speculative testing we also have a test that runs 14 m cpp-beam_search_causal_lm-windows.

So it's comparable and not critical i think

Reply:
Thanks, I am fine with it then